### PR TITLE
[RW-5114][risk=no] Update hierarchy page 

### PR DESCRIPTION
**Pull request by dolbeew**
- Style updates for hierarchy view for V2 UI
- Add store for opening/closing sidebar from other components, use for 'Finish & Review' button
- Move `AttributesPageV2` component to `SelectionList`
- Add shared function `saveCriteria` for adding current criteria selections to cohort

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] I have run and tested this change locally

### CODE REVIEW

File: ui/src/app/cohort-search/list-search-v2/list-search-v2.component.tsx [hunk 1]
```diff
@@ -411,7 +411,12 @@ export const ListSearchV2 = withCurrentWorkspace()(
             </tbody>
           </table>}
           {!standardOnly && !displayData.length && <div>No results found</div>}
-          <Button type='primary' style={{borderRadius: '5px', float: 'right', marginTop: '1rem'}}>Finish & Review</Button>
+          <Button type='primary'
+                  style={{borderRadius: '5px', float: 'right', marginTop: '1rem'}}
+                  disabled={selectedIds.length === 0}
```
Comment (NehaBroad) [inline ui/src/app/cohort-search/list-search-v2/list-search-v2.component.tsx:416]: Should we add test for selectionsIds being null or undefined?

File: ui/src/app/cohort-search/selection-list/selection-list.component.tsx [hunk 2]
```diff
@@ -310,111 +329,143 @@ export class SelectionListModalVersion extends React.Component<Props> {
...
+        criteria[0].domainId !== DomainType.PHYSICALMEASUREMENT.toString()
+        && criteria[0].domainId !== DomainType.PERSON.toString();
+    }
+
+    render() {
+      const {back, criteria} = this.props;
+      const {attributesSelection, modifierButtonText, showModifiersSlide} = this.state;
+      return <div>
+        <FlexRow style={styles.navIcons}>
+          {(attributesSelection || showModifiersSlide) &&
```
Comment (NehaBroad) [inline ui/src/app/cohort-search/selection-list/selection-list.component.tsx:427]: nit we can move this in a function as we are using this condition twice (at 442) attributesSelection || showModifiersSlide

File: ui/src/app/cohort-search/tree-node/tree-node.component.tsx [hunk 3]
```diff
@@ -292,7 +292,9 @@ export class TreeNode extends React.Component<TreeNodeProps, TreeNodeState> {
       setAttributes} = this.props;
     const {children, error, expanded, hover, loading, searchMatch} = this.state;
     const nodeChildren = domainId === DomainType.PHYSICALMEASUREMENT.toString() ? node.children : children;
-    const selected = selectedIds.includes(this.paramId) || groupSelections.includes(parentId);
+    const selected = serverConfigStore.getValue().enableCohortBuilderV2
+      ? currentCohortCriteriaStore.getValue().some(crit => crit.parameterId === this.paramId)
+      : selectedIds.includes(this.paramId) || groupSelections.includes(parentId);
```
Comment (NehaBroad) [inline ui/src/app/cohort-search/tree-node/tree-node.component.tsx:297]: Nit: can you please add brackets, because of so many conditions its a little difficult to read
